maggy/core/environment/hopsworks.py

```diff
@@ -0,0 +1,258 @@
+#
+#   Copyright 2021 Logical Clocks AB
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+import json
+import os
+
+import hsfs
+from hops import constants
+from hops import hdfs as hopshdfs
+from hops import util as hopsutil
+from hops.experiment_impl.util import experiment_utils
+from maggy import tensorboard
+from maggy import util
+from maggy.core.environment.base import BaseEnv
+
+
+class HopsworksEnv(BaseEnv):
+    """
+    This class implements AbstractEnvironment.
+    The methods implemented mainly recall the libraries developed on maggy.
+    """
+
+    def __init__(self, *args):
+        self.constants = constants
```

Comment:
When you initialize the environment in the singleton class, I don't see you passing the constants, so I am surprised how this works.

Reply:
They are imported